pinc/project_states.inc

```diff
@@ -40,262 +40,282 @@ class ProjectState
     }
 }
 
-global $project_states;
-
-function declare_project_state(
-    $constant_name,
-    $constant_value,
-    $medium_label,
-    $long_label,
-    $forum,
-    $phase,
-    $star_metal
-) {
-    global $project_states;
-
-    define($constant_name, $constant_value);
-
-    $project_states[$constant_value] = new ProjectState(
+class ProjectStates
+{
+    private static $states = [];
+
+    private static function declare_project_state(
+        $constant_name,
+        $constant_value,
         $medium_label,
         $long_label,
         $forum,
         $phase,
         $star_metal
-    );
-}
-
-function get_medium_label_for_project_state($state)
-{
-    global $project_states;
-    return $project_states[$state]->medium_label ?? '';
-}
-
-function project_states_text($state)
-{
-    global $project_states;
-    return $project_states[$state]->long_label ?? '';
-}
+    ) {
+        define($constant_name, $constant_value);
+
+        self::$states[$constant_value] = new ProjectState(
```

Comment:
If we follow the existing Activity/Activities pattern of having the `ProjectState` constructor assign itself to `ProjectStates` we avoid a lot of this code churn while still getting all of the benefits of having the "container" `ProjectStates` class. And while I don't really love that pattern, it is used a fair amount in our code and I'd almost rather keep it for consistency.

Reply:
Hmmm...